Ilya/NetflixRoulette/netflix-roulette/src/redux/moviesSagas.ts

```diff
@@ -0,0 +1,77 @@
+import {
+    MoviesActionTypes,
+    RequestMoviesAction,
+    requestMoviesSuccess,
+    requestMoviesError,
+    RequestMovieByIdAction,
+    requestMovieByIdSuccess,
+    requestMovieByIdError
+} from "./moviesActions";
+import { all, call, put, takeLatest } from 'redux-saga/effects';
+import queryString from "query-string";
+
+function* requestMoviesSaga(action: RequestMoviesAction) {
+    let search: string = action.payload;
+    let offset = 0;
+
+    if (search === "") {
+        search = "sortBy=release_date&offset=0";
+    } else {
+        let parsed = queryString.parse(search);
+        if ('page' in parsed) {
+            offset = (Number(parsed.page) - 1) * 9;
+            delete parsed.page;
+            parsed.offset = String(offset);
+            search = queryString.stringify({ ...parsed });
+            console.log(search);
```

Comment:
you don't need it here, because you still will do it on 28 line

Reply:
Done